app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Admin::UsersController < ApplicationController
+  before_filter :check_if_admin
+
+  def index
+    @users = User.all
+  end
+
+  def destroy
+    @user = User.find(params[:id])
```

Comment:
Flow control + flashes. Imagine if a user deleted their own account the moment before an admin tried to. Same thing on other things like playlists.